### ISSUE TITLE
cl.oauth_callback can't work

### ISSUE THREAD

**Issue by asd332689912** [Chainlit/chainlit#1001]
**Describe the bug**
I defined a third-party ssoprovider myself and added the provider using from chainlit.oauth_providers import providers. Jumping to the login page has been implemented. However, when I clicked to log in, my custom oauth_callback() method failed to receive the request, which was automatically sent to chainlit's own oauth_callback(), preventing me from changing the logon logic.

**To Reproduce**
1.my provider:
`class FooBarProvider(OAuthProvider):
    id="your-id"
    env=["YOUR_ENV_VAR_NAMES"]

    authorize_url=f"https://get.your/auth_url"
    token_url=f"https://get.your/token_url"
    
    def __init__(self):
        self.client_id = os.environ.get("YOUR_ENV_VAR_NAMES")
    
    async def get_token(self, code: str, url: str) -> str:
        payload = { "foo": self.client_id }
        async with httpx.AsyncClient() as client:
            response = await client.post(
                self.token_url,
                data=payload
            )`
2.inject_custom_auth.py
`......
def add_custom_oauth_provider(provider_id: str, custom_provider_instance):
    if custom_oauth_enabled() and not provider_id_in_instance_list(provider_id):
        providers.append(custom_provider_instance)
        print(f"Added provider: {provider_id}")
    else:
        print(f"Custom OAuth is not enabled or provider {provider_id} already exists")`
3.injection  app.py
`from cookbook.auth.foobar_oauthprovider import FooBarProvider
from cookbook.auth.inject_custom_auth import add_custom_oauth_provider

add_custom_oauth_provider("foobar", FooBarProvider())

@cl.oauth_callback
def oauth_callback(
  provider_id: str,
  token: str,
  raw_user_data: Dict[str, str],
  default_user: cl.User,
) -> Optional[cl.User]:
  return default_user  `

**Expected behavior**
oauth_callback() can recive callback and modify logic

**Screenshots**
![qqqq](https://github.com/Chainlit/chainlit/assets/45085652/572b4325-c77b-4058-8300-e55384e5b50b)
![1715873091180](https://github.com/Chainlit/chainlit/assets/45085652/af0743cc-6977-411e-b53f-d0fbda83f7f9)

**Desktop (please complete the following information):**

- OS: win11
- Browser chrome
- Version chainlit-main

**Comment by nestorcolt-clv** [comment]
I'm also having this issue, I thought it was Cognito at first. Apparently is not.

**Comment by asd332689912** [comment]
@nestorcolt-clv Yes,Yes, I thought there was something wrong with my program at first, but after checking it, it didn't work. I wonder if there is a way to solve this problem,I need support our SSO(Oauth2) systemctl.

**Comment by dokterbob** [comment]
@asd332689912 @nestorcolt-clv Are you still experiencing this issue withe the current chainlit version?
